src/lib/storage/column_iterables.hpp

```diff
@@ -92,7 +92,10 @@ class ColumnIterable {
    */
   template <typename Container>
   void materialize_values(Container& container) const {
-    for_each([&](const auto& value) { container.push_back(value.value()); });
+    DebugAssert(container.empty(), "Trying to materialize into a non-empty container");
+    container.resize(_self()._on_size());
```

Comment:
Just to be sure I understood last week's diskussion: reserve() cannot be used here because it will not effect size and thus iterators would not work afterwards.

Reply:
We would have to use push_back then, which has two steps. Overwriting is faster.